src/WebJobs.Script.WebHost/Security/Authentication/AuthenticationLevelHandler.cs

```diff
@@ -105,6 +108,8 @@ internal class AuthenticationLevelHandler : AuthenticationHandler<Authentication
             return AuthorizationLevel.Anonymous;
         }
 
-        private static bool HasMatchingKey(IDictionary<string, string> secrets, string keyValue) => secrets != null && secrets.Values.Any(s => Key.SecretValueEquals(s, keyValue));
+        private static bool HasMatchingKey(IDictionary<string, string> secrets, string keyValue, string keyName = null)
```

Comment:
out of curiosity, when is the keyName overload used?

Reply:
Great question :)

This is not used *yet*. Kept this there because it will be needed for WebHooks and system keys (eventually).